### ISSUE TITLE
3.1.0 bug, Assigning to rvalue

### ISSUE THREAD

**Issue by drcmda** [gaearon/react-hot-loader#672]
Fixing [#662](https://github.com/gaearon/react-hot-loader/issues/662) must've caused something else, updating from beta-7 to 3.1.0 gave the following: 

```
ERROR in ./core/renderer.js
Module parse failed: /Users/.../node_modules/babel-loader/lib/index.js??ref--0!/Users/.../core/renderer.js Assigning to rvalue (121:19)
You may need an appropriate loader to handle this file type.
| 
|   _proto.__resizeHandler__REACT_HOT_LOADER__ = function __resizeHandler__REACT_HOT_LOADER__() {
|     const canvas = errUndefined`Factory was initialized without canvas` = this.props;
|     return this.resize();
|   };
```

It seems to add `this.props` to a random classes default parameter thereby causing invalid code. The class has no relation to react.

Original code:

```js
function errUndefined(text) {
    throw new Error(text)
}

export default class Renderer {
    constructor(canvas = errUndefined`Renderer initialized w/o canvas`, options = {}) {
        this.resizeHandler = () => this.resize()
        ...
```

**Comment by neoziro** [comment]
OK, also I think it is not a good idea because there is a lot of cases that will not work:

```js
class Foo extends React.Component {
  constructor(props) {
    super(props)
    const foo = 'bar'
    this.arrow = () => console.log(foo)
  }
}
```

I will revert all logic of extracting arrow functions from constructor.

**Comment by neoziro** [comment]
Fixed in v3.1.1
